cloud_init.rb

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env ruby
+
+# Copyright (C) 2015 Oregon State University
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.
+
+require 'yaml'
+
+host = ENV['INSTANCE_NAME'] || ''
```

Comment:
It would look something like this:

```ruby
require 'yaml'

def cloud_init
  host = ENV['INSTANCE_NAME'] || ''
  ...
end

if __FILE__ == $0
  cloud_init
end
```

Then you can pull in the file and do tests on the 'cloud_init' method.